addons/sale/models/payment.py

```diff
@@ -63,32 +63,29 @@ def _set_transaction_pending(self):
         super(PaymentTransaction, self)._set_transaction_pending()
 
         for record in self:
-            sales_orders = record.sale_order_ids.filtered(lambda so: so.state == 'draft')
-            sales_orders.force_quotation_send()
-
+            sales_orders = record.sale_order_ids.filtered(lambda so: so.state in ['draft', 'sent'])
             if record.acquirer_id.provider == 'transfer':
                 for so in sales_orders:
                     so.reference = record._compute_sale_order_reference(so)
+            sales_orders.with_context(mark_so_as_sent=True).send_order_confirmation_mail(transaction=record)
 
     @api.multi
     def _set_transaction_authorized(self):
         # Override of '_set_transaction_authorized' in the 'payment' module
         # to confirm the quotations automatically.
         super(PaymentTransaction, self)._set_transaction_authorized()
-        sales_orders = self.mapped('sale_order_ids').filtered(lambda so: so.state == 'draft')
-        sales_orders.force_quotation_send()
-        sales_orders = self.mapped('sale_order_ids').filtered(lambda so: so.state == 'sent')
+        sales_orders = self.mapped('sale_order_ids').filtered(lambda so: so.state in ['draft', 'sent'])
         for so in sales_orders:
             # For loop because some override of action_confirm are ensure_one.
             so.action_confirm()
+        for transaction in self:
+            sales_orders.send_order_confirmation_mail(transaction=transaction)
```

Comment:
It means you are sending mail on tx X sale orders as you are basically doing a double loop.